js/components/addEditBookmark.js

```diff
@@ -12,7 +11,7 @@ const KeyCodes = require('../constants/keyCodes')
 const siteTags = require('../constants/siteTags')
 const siteUtil = require('../state/siteUtil')
 
-class AddEditBookmark extends ImmutableComponent {
+class AddEditBookmark extends React.Component {
```

Comment:
Thanks for the patch. 

Please keep this as ImmutableComponent, and then just determine in render if it's valid or not and render accordingly.  Also I think this only styles that it's invalid currently and not stops it from submitting it. 

For any visual changes also pls include a screenshot, thanks.